src/phel/json.phel

```diff
@@ -1,33 +1,42 @@
 (ns phel\json
-  (:use \Exception))
+  (:use \RuntimeException))
 
-(defn valid-key? [v]
+(defn valid-key?
+  "Checks if `v` is a valid JSON key or can be converted to a JSON key."
+  [v]
   (or (int? v) (float? v) (symbol? v) (keyword? v) (string? v)))
 
+(declare encode-value [x])
+
+(defn- encode-value-iterable [x]
+  (let [arr (php/array)]
+    (foreach [k v x]
+      (when-not (valid-key? k)
+        (throw (php/new RuntimeException "Key can only be an integer, float, symbol, keyword or a string.")))
+      (php/aset arr (encode-value k) (encode-value v)))
+    arr))
+
 (defn encode-value [x]
+  "Convert a Phel data type to a 'json compatible' value."
   (cond
-    (php/is_iterable x)
-    (let [arr (php/array)]
-      (foreach [k v x]
-        (when-not (valid-key? k)
-          (throw (php/new Exception "Key can only be an integer, float, symbol, keyword or a string.")))
-        (php/aset arr (encode-value k) (encode-value v)))
-      arr)
-    (symbol? x) (str (php/-> x (getName)))
-    (keyword? x) (str (php/-> x (getName)))
+    (php/is_iterable x) (encode-value-iterable x)
+    (symbol? x) (name x)
+    (keyword? x) (name x)
     (float? x) (str x)
     true x))
 
 (defn encode [value & [{:flags flags :depth depth}]]
+  "Returns the JSON representation of a value."
   (let [flags (or flags 0)
         depth (or depth 512)]
-    (when (php/is_resource value) (throw (php/new Exception "Value can be any type except a resource.")))
-    (when-not (int? flags) (throw (php/new Exception "Flags must be an integer.")))
-    (when-not (int? depth) (throw (php/new Exception "Depth must be an integer.")))
-    (when-not (> depth 0) (throw (php/new Exception "Depth must be greater than zero.")))
+    (when (php/is_resource value) (throw (php/new RuntimeException "Value can be any type except a resource.")))
```

Comment:
What about instead of throwing a RuntimeException, return a [JsonException](https://www.php.net/manual/en/class.jsonexception.php) as PHP does?

Reply:
Good idea. I will change it 👍